flytekit/core/python_auto_container.py

```diff
@@ -40,6 +49,8 @@ def __init__(
         environment: Optional[Dict[str, str]] = None,
         task_resolver: Optional[TaskResolverMixin] = None,
         secret_requests: Optional[List[Secret]] = None,
+        pod_template: Optional[PodTemplate] = None,
+        pod_template_name: Optional[str] = None,
```

Comment:
Should we briefly explore name changes here? `pod_template` and `pod_template_name` are not very descriptive and certainly don't capture a client-side `PodTemplate` application vs server-side k8s resource `PodTemplate` configuration. Thoughts?

Reply:
yeah it is very confusing now, but honestly I don't have a better idea on my mind. I also feel they should be `one of`. If users can set both, it will be more confusing.

Reply:
Still needs discussion!